protocols/raft/src/main/java/io/atomix/protocols/raft/metrics/SnapshotReplicationMetrics.java

```diff
@@ -0,0 +1,44 @@
+package io.atomix.protocols.raft.metrics;
+
+import io.prometheus.client.Gauge;
+
+public class SnapshotReplicationMetrics extends RaftMetrics {
+  private static final String NAMESPACE = "atomix";
+  private static final String PARTITION_GROUP_NAME_LABEL = "partitionGroupName";
+  private static final String PARTITION_LABEL = "partition";
+
+  private static final Gauge COUNT =
+      Gauge.build()
+          .namespace(NAMESPACE)
+          .labelNames(PARTITION_GROUP_NAME_LABEL, PARTITION_LABEL)
+          .help("Count of ongoing snapshot replication")
+          .name("pending_snapshot_count")
+          .register();
+  private static final Gauge DURATION =
+      Gauge.build()
+          .namespace(NAMESPACE)
+          .labelNames(PARTITION_GROUP_NAME_LABEL, PARTITION_LABEL)
+          .help("Approximate duration of replication in milliseconds")
+          .name("pending_snapshot_milliseconds")
```

Comment:
The name `pending_snapshot_milliseconds` is confusing. Better something like `snapshot_replication_duration_milliseconds`.